cajita/src/Cajita_SparseDimPartitioner.hpp

```diff
@@ -0,0 +1,640 @@
+#ifndef CAJITA_SPARSEDIMPARTITIONER_HPP
+#define CAJITA_SPARSEDIMPARTITIONER_HPP
+
+#include <Cajita_Partitioner.hpp>
+#include <Cajita_SparseIndexSpace.hpp>
+#include <Kokkos_Core.hpp>
+
+#include <array>
+#include <vector>
+
+#include <mpi.h>
+
+namespace Cajita
+{
+//---------------------------------------------------------------------------//
+template <typename MemorySpace, typename ExecSpace,
```

Comment:
We should template on `DeviceType` instead, then extract the memory and execution space from it

Reply:
Sounds great. I'll change that.